src/pie_utils/processor/document/split_into_partitions.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+
+import json
+import logging
+import re
+from typing import Any, Callable, Iterable, Iterator, Match
+
+from pytorch_ie.annotations import Span
+
+from pie_utils.span.slice import have_overlap
+from pie_utils.statistics import WithStatistics
+
+from ..document import DocumentWithPartition
+
+logger = logging.getLogger(__name__)
+
+
+def get_partitions_with_matcher(
+    document: DocumentWithPartition,
+    matcher: Callable[[str], Iterable[Match]],
+    label_group_id: int | None = None,  # = 1,
+    label_whitelist: list[str] | None = None,
+    skip_initial_partition: bool = False,  # = True
+) -> Iterator[Span]:
+    """Spans are created starting with the beginning of matching entries end ending with the start
+    of the next matching one or the end of the document.
+
+    Entries with xml node names that are not in label_whitelist, if it is set, will not be
```

Comment:
this is not about xml nodes (you can sue matcher created from any regex)

Reply:
fixed in [a9eb530](https://github.com/ArneBinder/pie-utils/pull/21/commits/a9eb5309bb21ae4a6ace9e3dbcadab2869a1c537)